### ISSUE TITLE
Seriously, this API library needs some attention.

### ISSUE THREAD

**Issue by shhh9** [alpacahq/alpaca-trade-api-python#269]
The momentum of development and bug fixes on this API library is just not up to par with the expectations. We ran into issues that were opened months ago without fixes.

For example: This bug, throwing "TypeError: 'NoneType' object is not iterable" when no data from polygon is returned for the given ticker and date range, has been reported since Jul 15, 2019 (https://github.com/alpacahq/Alpaca-API/issues/78). It has been reported multiple times if you do a simple Google search, and yet, at this time of writing, a year later, is still not fixed. The developers kept brushing it off as a polygon issue, or closing it without really fixing anything. It is this library that is not handling empty results gracefully. And there are so many other bugs like this, with simple fixes, and yet they are lingering around for a long time.

If Alpaca wants people to seriously trade with real money, this needs to addressed fast. We need to see serious commitment to quality and reliability on the client libraries before we can commit meaningful assets to the platform.

**Comment by umitanuki** [comment]
The reported issue you reference was a question about the available data range that Polygon provides. You seem to be insisting about the handling of empty data in the library. They are different issues. It's a fair report the library should not report such exception when the response is not empty, and I'm open to change the behavior. Just to clarify, with that change the client library still needs to handle None as a returned value from the function.

If things don't seem fast enough from your view, we apologize. Whether we are sleeping for the last 18 months? Absolutely not. There are tons of work that lots of people are waiting to be done as well and we are trying to be as fast as possible. The library has been updated a lot to address many issues as well, while we try as carefully as possible to provide the best balance between stability (in terms of behavior) and addition of new stuff. If you feel frustrated, remember, this is an open source library with Apache 2 license, and you are more than welcome to contribue any small PRs as many others do.

**Comment by shhh9** [comment]
I apologize for the tone in my previous message. I do hope more could be done to improve the documentation, quality and reliability of the client libraries as they are critical to adoption. I will close this one and reopen a new issue for the bug I mentioned.
